### PR TITLE
Remove the `asgdLclVar` analysis from liveness.

### PR DESCRIPTION
**Pull request by pgavlin**
This analysis is not necessary for correctness and does not appear to
have any impact on the generated code: in particualr, internal assembly
diffs show no changes with this analysis disabled.

For example, consider the following statement:

    `(asg (lclVar V0) (add (lclVar V0) (icon 4)))`

Under the `asgdLclVar` analysis, the visit of the lclVar node that is an
operand to the add would mark the lclVar node that is the target of the
assignment with `GTF_VAR_USEDEF` and exit without adding V0 to the
current set of used lclVars. Instead, the visit of the lclVar node that
is the target of the assignment would add V0 to both the set of used
lclVar nodes and the set of defined lclVar nodes.

With this analysis disabled, the visit of the lclVar node that is an
operand to the add will add V0 to the set of used lclVar nodes and the
visit of the lclVar node that is the target of the assignment will add
V0 to the set of defined lclVar nodes. In both cases, the result is the
same.

### CODE REVIEW

**Comment by pgavlin** [comment]
@CarolEidt @dotnet/jit-contrib PTAL

**Comment by CarolEidt** [comment]
So - there were no asm diffs even for the legacy 32-bit JIT? I thought there were some late optimizations to use RMW addressing modes in these cases (though I don't believe I ever actually observed that myself).

**Comment by pgavlin** [comment]
@CarolEidt: correct. IIUC the legacy 32-bit JIT uses `GTF_VAR_USEASG` for the optimizations you're thinking of.

**Comment by pgavlin** [comment]
In fact, after this change, `GTF_VAR_USEDEF` is never set. I will send a follow-up change to remove the flag and its associated code.

File: src/jit/liveness.cpp [hunk 1]
```diff
@@ -22,19 +22,12 @@
...
-        assert(tree->OperIsLocalField());
-        lclNum = tree->gtLclFld.gtLclNum;
-    }
+    assert((tree->OperIsLocal() && (tree->OperGet() != GT_PHI_ARG)) || tree->OperIsLocalAddr());
 
+    const unsigned lclNum = tree->gtLclNum;
     assert(lclNum < lvaCount);
-    LclVarDsc* const varDsc = lvaTable + lclNum;
+
+    LclVarDsc* const varDsc = &lvaTable[lclNum];
```
Comment (mikedn) [inline src/jit/liveness.cpp]: @pgavlin I would have added a `lvaGetDesc(GenTree*)` function to the compiler :)

**Pull request merged**